xmtp_content_types/src/lib.rs

```diff
@@ -2,13 +2,74 @@ pub mod group_updated;
 pub mod membership_change;
 pub mod text;
 
+use diesel::{
+    backend::Backend,
+    deserialize::{self, FromSql, FromSqlRow},
+    expression::AsExpression,
+    serialize::{self, IsNull, Output, ToSql},
+    sql_types::Integer,
+    sqlite::Sqlite,
+};
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 use xmtp_proto::xmtp::mls::message_contents::{ContentTypeId, EncodedContent};
 
+/// ContentType and their corresponding string representation
+/// are derived from the `ContentTypeId` enum in the xmtp-proto crate
+/// that each content type in this crate establishes for itself
+#[repr(i32)]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, FromSqlRow, AsExpression)]
+#[diesel(sql_type = diesel::sql_types::Integer)]
```

Comment:
Seems a bit off to have the DB specific code in the `xmtp_content_types` crate, but the alternative would be to have a replica of this enum in `storage/encrypted_store/group_message` sort of like this example of `GroupMessageKind`

https://github.com/xmtp/libxmtp/blob/eccce0610585310713951a995200bb6e8dfd8bf6/xmtp_mls/src/storage/encrypted_store/group_message.rs#L49-L55

The benefit of how I have it organized now is that when you add a new content type, you do not have to update the same enum in two different places. But I'm open to feedback here.

Reply:
moved to storage/encrypted_store/group_message to simplify